sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -1018,6 +1046,52 @@ uint32_t ProgramManager::getDeviceLibReqMask(const RTDeviceBinaryImage &Img) {
     return 0xFFFFFFFF;
 }
 
+ProgramManager::KernelArgMask ProgramManager::getEliminatedKernelArgMask(
+    OSModuleHandle M, const context &Context, pi::PiProgram NativePrg,
+    const string_class &KernelName, bool KnownProgram) {
+  if (m_UseSpvFile && M == OSUtil::ExeModuleHandle)
+    return {};
+
+  {
+    std::lock_guard<std::mutex> Lock(MNativeProgramsMutex);
```

Comment:
@romanovvlad, I think this illustrates the benefits of storing masks in the int header and having them as compile-time constant, rather than looking up dynamically like we do here. With compile-time constant, the mask can be captured by `StoreLambda` in handler.hpp and propagated to the place of kernel invocation.

@sergey-semenov - can you please add a TODO? I think this part of design should be discussed again.

Reply:
They would not be compile time constants provided we can have several images and we do not know which mask to pick at compile time. Also if we go with headers approach we will need to move binary selection to the CG construction time.
But I agree that lookup would be more cheep in this case.

Reply:
well, masks themselves will be constants, but indeed, there can be runtime code which selects one of many.